### PR TITLE
Increase robustness of EU-DEMO build and fix closure problem

### LINKED ISSUE #1078: Some closure inconsistencies may be causing bugs or cause bugs in future.
## Describe the bug

When a wire is closed in the primitive form, I assume this means that its start and end points are coincident to within some tolerance. I do not know if FreeCAD automatically "snaps" them to be mathematically identical, but I suspect not.

I've noticed that when discretising a closed wire with `byedges=False` the resulting numpy array (under the hood) is not closed in such a way that when passed into `Coordinates` it is considered open (i.e. that `np.allclose(points[0], points[-1], rtol=0, atol=EPS) == False`

If I pass this "open" set of `Coordinates` back into `make_polygon`, the resulting wire is indeed closed (even with `make_polygon(coordinates, closed=False)`.

## Steps to reproduce

Steps to reproduce the behaviour:

```python
from bluemira.geometry.parameterisations import TripleArc
from bluemira.builders.EUDEMO.tools import varied_offset

wire = TripleArc().create_shape()
coordinates = wire.discretize(ndiscr=100, byedges=False)

offset_wire = varied_offset(wire, 0.2, 0.6, 60, 160, num_points=200)

print(offset_wire.is_closed())

offset_coordinates = offset_wire.discretize(ndiscr=100, byedges=False)
print(offset_coordinates.closed)
```

## Expected behaviour

I expect the discretisation of closed wire to be closed within machine precision. Alternatively I expect that our "machine precision for geometry" be the same across FreeCAD and discretised representations.

## Additional context
I'm investigating the various issues we are seeing in the EU-DEMO build where occasionally we get open wires, disjointed faces, reconstruction errors, etc. I'm not sure that this is related, but I'm trying to bomb-proof stuff as much as possible!

@ivanmaione FYI

### PR DESCRIPTION
**Pull request by CoronelBuendia**
## Linked Issues

closes #1078 

## Description

Switches offset method in thermal shield to "intersect" which seems to be more robust in this case.

Adds some geometry closed checks.

## Interface Changes

N/A

## Checklist

I confirm that I have completed the following checks:

- [x] Tests run locally and pass `pytest tests --reactor`
- [x] Code quality checks run locally and pass `flake8` and `black .`
- [x] Documentation built locally and checked `sphinx-build -W documentation/source documentation/build`

### CODE REVIEW

File: bluemira/codes/_freecadapi.py [hunk 1]
```diff
@@ -660,6 +660,9 @@ def discretize(w: Part.Wire, ndiscr: int = 10, dl: float = None):
     # discretization points array
     output = w.discretize(ndiscr)
     output = vector_to_numpy(output)
+
+    if w.isClosed():
```
Comment (ivanmaione) [inline bluemira/codes/_freecadapi.py:664]: I think this "trick" should be moved into the `discretize` method of BluemiraWire. `_freecadapi` methods should return, if possible, what is expected doing the same operation in FreeCAD.

File: bluemira/geometry/tools.py [hunk 2]
```diff
@@ -487,6 +487,10 @@ def _offset_wire_discretised(
     """
     from bluemira.geometry._deprecated_offset import offset_clipper
 
+    if not open_wire:
```
Comment (ivanmaione) [inline bluemira/geometry/tools.py:490]: Not sure how the `open_wire` parameter combines with the note "Can only handle closed wires". A little bit confusing for me. Probably renaming it as "force_closure" (or similar) would be more explanatory? Similar comment for the signature of `offset_wire`.
Moreover, the method should raise an error if the wire is not closed and, in this case, `open_wire` is `True`.

**Review by ivanmaione — CHANGES_REQUESTED** [review]

File: bluemira/codes/_freecadapi.py (hunk 1 above)
Comment (CoronelBuendia) [reply bluemira/codes/_freecadapi.py:664]: Yeah, that makes sense - will do

File: bluemira/geometry/tools.py (hunk 2 above)
Comment (CoronelBuendia) [reply bluemira/geometry/tools.py:490]: Agree to clean this up... I probably need to update my environment as I ran this 10 times without any failure on my machine at work. It fails first time in the CI with an error I've never seen :/

**Comment by hsaunders1904** [comment]
Sorry, new error: `bluemira.geometry.error.NotClosedWire: The open boundary has not been closed.`. [Error log](https://github.com/Fusion-Power-Plant-Framework/bluemira/files/8741705/error-log2.txt).

I haven't been able to reproduce outcome 3 in https://github.com/Fusion-Power-Plant-Framework/bluemira/issues/1049 though - so that's good.

File: bluemira/codes/_freecadapi.py (hunk 1 above)
Comment (CoronelBuendia) [reply bluemira/codes/_freecadapi.py:664]: I note that the same trick is already used in `discretize_by_edges` in `_freecadapi.py`

Comment (ivanmaione) [reply bluemira/codes/_freecadapi.py:664]: _youthful mistakes_ :) I still support my first comment idea, so I would propose to change also `discretize_by_edges`.  The problem probably resides in the fact that there is no counterpart in `tools.py`.

Comment (CoronelBuendia) [reply bluemira/codes/_freecadapi.py:664]: Farmed this off to #1085 for now. Probably needs a broader treatment rather than a piecemeal approach.

**Review by ivanmaione — APPROVED** [review]

**Pull request merged**